feat: more useful error messages

balloonpopper opened the pull request:
The current error messages are not as developer friendly as they could be. This addresses that as per #108 
It does not however fix the other fault mentioned in that issue where the filename is not mentioned as part of the error message.

Before:
 2022-2-22T173259 (E)	Errors in file Argument type did not match descriptor for command "camera_shift"
Argument 1  is of type 4. Expected 2,3

After:
 2022-2-22T173259 (E)	Errors in file Argument type did not match descriptor for command "camera_shift"
Argument 1 ("-100") is of type TYPE_STRING. Expected [TYPE_INT|TYPE_REAL]

FYI @dploeger @StraToN @BHSDuncan

dploeger reviewed (approved):
And there it is already" Thanks a lot! Very helpful. 🎁

BHSDuncan commented on addons/escoria-core/game/core-scripts/esc/types/esc_command_argument_descriptor.gd:
```diff
@@ -2,6 +2,15 @@
+# As the get_type command was deprecated with Godot 2.x w we need a way to determine
+# variable types. Ideally these wouldn't be hardcoded but there's no GDScript 3.x command to
+# turn a type back to its name.
+var godot_type_list = ["TYPE_NIL", "TYPE_BOOL", "TYPE_INT", "TYPE_REAL",  "TYPE_STRING", \
```
Minor nitpick/comment/not married to it, but, unless this list can be added to dynamically, maybe this should be marked as `const`? I realize that arrays in GDScript are passed by reference anyway, but for the sake of intent (a la pseudo-private prefixes), maybe this makes some sense. Other than that, this PR looks good.

BHSDuncan replied:
If you can CONSTANT_CASE the name here, I think we're good!

balloonpopper replied:
Made a couple of changes here in addition to CONSTANT_CASE. I've replaced the "|" with " or ", removed "TYPE_" off all the types,  and lower-cased all the types to make it more readable. New error example:
Argument 0 ("-300") is of type string . Expected [ int or real ]

balloonpopper commented:
Made a couple of changes here in addition to CONSTANT_CASE. I've replaced the "|" with " or ", removed "TYPE_" off all the types, and lower-cased all the types to make it more readable. New error example:
```
Argument 0 ("-300") is of type string . Expected [ int or real ]
```

BHSDuncan commented:
Works for me! Thanks!

balloonpopper commented:
One more tiny update. I didn't like the additional space after the "type string ." before the "." - now fixed.
Argument 0 ("-300") is of type string. Expected [ int or real ]
